### ISSUE TITLE
Cryptic error message for nodes() when one does not import sequtils

### ISSUE THREAD

**Issue by disruptek** [ebran/grim#29]
Maybe we should look for sequtils and throw an error if it's not in the environment?

```
/home/adavidoff/git/unfunner/run.nim(69, 12) Error: type mismatch: got <iterator (): Node{.closure, locks: <unknown>.}>
but expected one of: 
iterator items(E: typedesc[enum]): E:type
  first type mismatch at position: 1
  required type for E: type enum
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(a: cstring): char
  first type mismatch at position: 1
  required type for a: cstring
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(a: string): char
  first type mismatch at position: 1
  required type for a: string
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(n: NimNode): NimNode
  first type mismatch at position: 1
  required type for n: NimNode
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(node: JsonNode): JsonNode
  first type mismatch at position: 1
  required type for node: JsonNode
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(p: Path): Edge
  first type mismatch at position: 1
  required type for p: Path
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items(pc: PathCollection): Path
  first type mismatch at position: 1
  required type for pc: PathCollection
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items[IX, T](a: array[IX, T]): T
  first type mismatch at position: 1
  required type for a: array[IX, T]
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items[T](a: openArray[T]): T
  first type mismatch at position: 1
  required type for a: openArray[T]
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items[T](a: seq[T]): T
  first type mismatch at position: 1
  required type for a: seq[T]
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items[T](a: set[T]): T
  first type mismatch at position: 1
  required type for a: set[T]
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}
iterator items[T](s: HSlice[T, T]): T
  first type mismatch at position: 1
  required type for s: HSlice[items.T, items.T]
  but expression '
let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
else:
  @newSeq(Natural(0))
iterator it`gensym37395217(): Node {.closure, gensym.} =
  for label`gensym37395218 in items(markers`gensym37395216):
    if not contains(nodeLabels(g), label`gensym37395218):
      continue
    for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
      let node = toMap(n`gensym37395219)
      if true:
        yield n`gensym37395219

it`gensym37395217' is of type: iterator (): Node{.closure, locks: <unknown>.}

expression: items(
  let markers`gensym37395216 = if len(newSeq(Natural(0))) == 0: nodeLabels(g)
  else:
    @newSeq(Natural(0))
  iterator it`gensym37395217(): Node {.closure, gensym.} =
    for label`gensym37395218 in items(markers`gensym37395216):
      if not contains(nodeLabels(g), label`gensym37395218):
        continue
      for n`gensym37395219 in values(g.nodeIndex[label`gensym37395218]):
        let node = toMap(n`gensym37395219)
        if true:
          yield n`gensym37395219
  
  it`gensym37395217)
```

**Comment by ebran** [comment]
Hi, cool to see that someone is using the library! 

Yeah, that's probably a good idea. This error is related to the seq/items template somehow. I don't quite figured out what was going on, but everything seems to work fine as long as sequtils is available...

**Comment by ebran** [comment]
sequtils was not exported in grim.nim, fixed by 22e1c51.
